plugins/modem-manager/fu-mm-device.c

```diff
@@ -1520,6 +1520,7 @@ fu_mm_device_init(FuMmDevice *self)
 {
 	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_UPDATABLE);
 	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_USE_RUNTIME_VERSION);
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_REQUIRE_AC);
```

Comment:
is this true?

Reply:
In theory you can upgrade it without external power, but the update takes a lot of time (~20 minutes) and if the battery runs out the modem can start to act weird. I don't know what the consequences would be if you lose power right during the flashing.

If the modem is bricked, you need to enter EDL mode to fix it which is hard:
- You have to connect 2 GPIO pins on the modem manually with some wire
- EDL mode is not really documented somewhere on how to fix it, so that's hard for people who don't know the internals of the modem.

That's why I purpose this patch, to avoid this situation as much as possible since this runs on phones now as well.
These devices have a battery that is way smaller than laptops and IMO we don't want people to upgrade modem firmwares when they are on the go without access to an external power source.

Reply:
@aleksander0m are you okay with adding this attribute for *all* MM devices? Or would you rather make it per-device with a `Flags = require-ac` quirk?

Reply:
I think it's a good idea to apply that to all modems, as they usually take quite a lot of time to upgrade.